external/storm-kafka/src/jvm/org/apache/storm/kafka/bolt/KafkaBolt.java

```diff
@@ -98,7 +98,11 @@ public void prepare(Map stormConf, TopologyContext context, OutputCollector coll
 
         //for backward compatibility.
         if(topicSelector == null) {
-            this.topicSelector = new DefaultTopicSelector((String) stormConf.get(TOPIC));
+            if(stormConf.containsKey(TOPIC)) {
+                this.topicSelector = new DefaultTopicSelector((String) stormConf.get(TOPIC));
+            } else {
+                throw new IllegalArgumentException("topic select should be specified");
```

Comment:
@darionyaphet You may want to have exception's message like below.
`'topic' should be specified in bolt's configuration`